packages/web3-eth-accounts/src/account.ts

```diff
@@ -53,10 +81,129 @@ export const privateKeyToAccount = (
 	const publicKeyString = `0x${publicKey.slice(2)}`;
 	const publicHash = sha3Raw(publicKeyString);
 	const publicHashHex = bytesToHex(publicHash);
-	const address = toChecksumAddress(publicHashHex.slice(-40)); // To get the address, take the last 20 bytes of the public hash
-	return { address, privateKey: stringPrivateKey, signTransaction, sign, encrypt };
+	return toChecksumAddress(publicHashHex.slice(-40)); // To get the address, take the last 20 bytes of the public hash
 };
 
+/**
+ * encrypt a private key given a password, returns a V3 JSON Keystore
+ * https://github.com/ethereum/wiki/wiki/Web3-Secret-Storage-Definition
+ */
+export const encrypt = async (
+	privateKey: HexString,
+	password: string | Buffer,
+	options?: CipherOptions,
+): Promise<Keystore> => {
+	if (!(isValidString(privateKey) || isBuffer(privateKey))) {
+		throw new InvalidPrivateKeyError(privateKey);
+	}
+
+	// if given salt or iv is a string, convert it to a Uint8Array
+	let salt;
+	if (options?.salt) {
+		salt = typeof options.salt === 'string' ? Buffer.from(options.salt, 'hex') : options.salt;
+	} else {
+		salt = randomBytes(32);
+	}
+	validateBytesInput(salt);
+
+	let iv;
+	if (options?.iv) {
+		iv = typeof options.iv === 'string' ? Buffer.from(options.iv, 'hex') : options.iv;
+	} else {
+		iv = randomBytes(16);
+	}
+	validateBytesInput(iv);
+
+	const kdf = options?.kdf ?? 'scrypt';
+
+	let derivedKey;
+	let kdfparams: ScryptParams | PBKDF2SHA256Params;
+
+	// derive key from key derivation function
+	if (kdf === 'pbkdf2') {
+		kdfparams = {
+			dklen: options?.dklen ?? 32,
+			salt: salt.toString('hex'),
+			c: options?.c ?? 262144,
+			prf: 'hmac-sha256',
+		};
+		derivedKey = pbkdf2Sync(
+			Buffer.from(password),
+			Buffer.from(salt),
+			kdfparams.c,
+			kdfparams.dklen,
+			'sha256',
+		);
+	} else if (kdf === 'scrypt') {
+		// FIXME: support progress reporting callback
```

Comment:
```suggestion
		// TODO: support progress reporting callback
```

Are we actually wanted to support a callback here? We are trying to get rid of all callback if possible

Reply:
This was a line from 1.x, this seems outdated so I am in favor of removing this